File: AppRemover/src/main/java/com/vs_unusedappremover/data/Plural.java
package com.vs_unusedappremover.data;

import android.content.res.Resources;
import android.util.Log;

import com.seppius.i18n.plurals.PluralResources;
import com.vs_unusedappremover.MyApplication;

public class Plural {
	
	private final Resources res;
	private PluralResources pluralRes;
	
	public Plural(Resources resources) {
		this.res = resources;
		try {
			this.pluralRes = new PluralResources(res);
		} catch (Exception e) {
			Log.e(MyApplication.TAG, "unable to create plural resources", e);
			this.pluralRes = null;
		}
	}
	
	public String format(int pluralResId, int quantity, Object...args) {
		if (pluralRes != null) {
			return pluralRes.getQuantityString(pluralResId, quantity, args);
		} else {
			return res.getQuantityString(pluralResId, quantity, args);
		}
	}
}

File: AppRemover/src/main/java/com/vs_unusedappremover/data/Plural.kt
package com.vs_unusedappremover.data

import android.content.res.Resources
import android.util.Log

import com.seppius.i18n.plurals.PluralResources
import com.vs_unusedappremover.MyApplication

class Plural(private val res: Resources) {

    private val pluralRes: PluralResources? by lazy {
        try {
            PluralResources(res)
        } catch (e: Exception) {
            Log.e(MyApplication.TAG, "unable to create plural resources", e)
            null
        }
    }

    fun format(pluralResId: Int, quantity: Int, vararg args: Any): String {
        val pluralRes = pluralRes
        return if (pluralRes != null) {
            pluralRes.getQuantityString(pluralResId, quantity, *args)
        } else {
            res.getQuantityString(pluralResId, quantity, *args)
        }
    }
}
